packages/resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -161,21 +162,74 @@ export class AwsEcsDetector implements ResourceDetector {
         AwsEcsDetector.DEFAULT_CGROUP_PATH,
         'utf8'
       );
-      const splitData = rawData.trim().split('\n');
-      for (const str of splitData) {
-        if (str.length > AwsEcsDetector.CONTAINER_ID_LENGTH) {
-          containerId = str.substring(
-            str.length - AwsEcsDetector.CONTAINER_ID_LENGTH
-          );
-          break;
-        }
+      const lines = rawData
+        .split('\n')
+        .map(s => s.trim())
+        .filter(Boolean);
+
+      // Pass 1: Prefer primary ECS pattern across all lines
+      for (const line of lines) {
+        const id = this._extractPrimaryEcsContainerId(line);
+        if (id) return id;
+      }
+
+      // Pass 2: Fallback to last-segment with strict allowed chars (hex + hyphen)
+      for (const line of lines) {
+        const id = this._extractLastSegmentContainerId(line);
+        if (id) return id;
+      }
+
+      // Pass 3: Legacy fallback to last 64 chars (Docker-style), keep existing behavior
+      for (const line of lines) {
+        const id = this._extractLegacyContainerId(line);
+        if (id) return id;
       }
     } catch (e) {
       diag.debug('AwsEcsDetector failed to read container ID', e);
     }
     return containerId;
```

Comment:
`containerId` is now redundant. I guess here you can now return undefined and remove this variable, or change `const id` to just be `containerId`.